### PR TITLE
Extract kde computing logic from matplotlib

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR moves core computation logic (in kde) from matplotlib module to core module in plot.

### CODE REVIEW

File: databricks/koalas/plot/core.py [hunk 1]
```diff
@@ -337,6 +338,43 @@ def get_fliers(colname, outliers, min_val):
...
+
+            sample_range = max_val - min_val
+            ind = np.linspace(min_val - 0.5 * sample_range, max_val + 0.5 * sample_range, 1000,)
+        elif is_integer(ind):
+            min_val, max_val = sdf.select(F.min(sdf.columns[-1]), F.max(sdf.columns[-1])).first()
+
+            sample_range = min_val - max_val
+            ind = np.linspace(min_val - 0.5 * sample_range, max_val + 0.5 * sample_range, ind,)
+        else:
+            ind = ind
```
Comment (ueshin) [inline databricks/koalas/plot/core.py:357]: nit: seems not needed.

File: databricks/koalas/plot/core.py [hunk 2]
```diff
@@ -337,6 +338,43 @@ def get_fliers(colname, outliers, min_val):
         return fliers
 
 
+class KdePlotBase:
+    @staticmethod
+    def get_ind(sdf, ind):
+        # 'data' is a Spark DataFrame that selects one column.
```
Comment (ueshin) [inline databricks/koalas/plot/core.py:344]: nit: `# 'sdf' is a ...`?

File: databricks/koalas/plot/core.py [hunk 3]
```diff
@@ -337,6 +338,43 @@ def get_fliers(colname, outliers, min_val):
...
+
+            sample_range = min_val - max_val
+            ind = np.linspace(min_val - 0.5 * sample_range, max_val + 0.5 * sample_range, ind,)
+        else:
+            ind = ind
+        return ind
+
+    @staticmethod
+    def compute_kde(sdf, bw_method=None, ind=None):
+        # 'data' is a Spark DataFrame that selects one column.
```
Comment (ueshin) [inline databricks/koalas/plot/core.py:362]: ditto.

**Review by ueshin — APPROVED** [review]
LGTM.

File: databricks/koalas/plot/core.py (hunk 2 above)
Comment (HyukjinKwon) [reply databricks/koalas/plot/core.py:344]: Ah, I don't know why I keep missing these stuff. Thanks @ueshin.

**Pull request merged**